eng/Directory.Build.Common.props

```diff
@@ -127,6 +127,8 @@
     <RequiredTargetFrameworks>net7.0;net6.0</RequiredTargetFrameworks>
     <!-- Also test net461 on Windows. -->
     <RequiredTargetFrameworks Condition="'$(OS)' == 'Windows_NT'">$(RequiredTargetFrameworks);net461</RequiredTargetFrameworks>
+    <!-- But only build snippets for the latest. -->
+    <RequiredTargetFrameworks Condition="'$(BuildSnippets)' == 'true'">net7.0</RequiredTargetFrameworks>
```

Comment:
This was actually an unexpected change I had to made once I fixed the first build error against `net7.0` - the main point of this commit. But I think this is what we want to do anyway. Snippets may use newer syntax, so why let older TFMs hobble us? It's also faster to run the Analyze step now.

Reply:
Interesting!  Do we have a precedent for providing samples for older supported versions of our libraries?  Curious.

Reply:
Would the older TFMs cause issues?    I'd have expected we'd always be compiling with `net7` and executing against the older platforms.    

I don't have a strong opinion, but it may be helpful to limit the new hotness features in snippets so that samples are more widely accessible.   Some of our larger customer segments are slower to adopt new syntax.

Reply:
This was necessary because Core samples used `ActivitySource` and friends, which isn't defined for net641. And because building snippets uses our existing `<TargetFrameworks>` to build (it's just the test projects) it does multi-targeting. That's what this change prevents.

As to @annelo-msft's question, I'd be surprised if we ever wrote any samples that weren't in newer targets. It's hypothetically possible (.NET "Core" can make breaking changes) but I think it not practically likely. Seems we could cross that bridge when we get to it.